### PR TITLE
Add --ignore-files option to exclude files from code coverage using path patterns

### PR DESCRIPTION
**Pull request by gogolon**
This makes it possible to exclude files which match glob patterns from the report, which was discussed in #427 and #313. The `.yaml` configuration part which was requested in #427 has not been implemented, but it can be done separately after this is merged.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

### CODE REVIEW

File: bin/format_coverage.dart [hunk 1]
```diff
@@ -124,50 +127,54 @@ Future<void> main(List<String> arguments) async {
 Environment parseArgs(List<String> arguments) {
   final parser = ArgParser();
 
-  parser.addOption('sdk-root', abbr: 's', help: 'path to the SDK root');
```
Comment (kevmoo) [inline bin/format_coverage.dart:127]: careful making unrelated changes. Makes reviews harder.

File: bin/format_coverage.dart [hunk 2]
```diff
@@ -66,6 +68,7 @@ Future<void> main(List<String> arguments) async {
   final hitmap = await HitMap.parseFiles(
     files,
     checkIgnoredLines: env.checkIgnore,
+    ignoreGlobs: env.ignoreFiles,
```
Comment (liamappelbe) [inline bin/format_coverage.dart:71]: Why not add this logic in the same place as `reportOn`? You could just add it to the `_getPathFilter` function in formatter.dart. Having the logic in two totally different places makes it a bit hard to read.

File: test/lcov_test.dart [hunk 3]
```diff
@@ -122,6 +124,37 @@ void main() {
...
+    test('formatLcov() excludes files matching glob patterns', () async {
+      final hitmap = await _getHitMap();
+
+      final resolver = await Resolver.create(packagePath: '.');
+      final res = hitmap.formatLcov(
+        resolver,
+        ignoreGlobs: {Glob('**/*.g.dart'), Glob('**/util.dart')},
+      );
+
+      expect(res, isNot(contains(p.absolute(_sampleGeneratedPath))));
```
Comment (liamappelbe) [inline test/lcov_test.dart:136]: I think you should be able to add an extra check for this file to all the other tests in this file: `expect(res, contains(p.absolute(_sampleGeneratedPath)));`. That way you're checking both the contains and doesn't contain cases.

At the moment I think these tests could pass accidentally by, for example, mistyping the `_sampleGeneratedPath`, since every test that checks it is checking that it's not in the report.

**Review by liamappelbe — COMMENTED** [review]
Looking good. Just need to tighten up the tests a bit.

File: test/lcov_test.dart (hunk 3 above)
Comment (gogolon) [reply test/lcov_test.dart:145]: I added this check as well as some additional checks in globs related tests to make the expected behaviour more explicit

**Review by liamappelbe — APPROVED** [review]
